src/legal-officer/LegalOfficerContext.tsx

```diff
@@ -147,32 +147,36 @@ export function LegalOfficerContextProvider(props: Props) {
 
     useEffect(() => {
         if(accounts !== null
+                && client
                 && axiosFactory !== undefined
                 && accounts.current !== undefined
                 && contextValue.dataAddress !== accounts.current.address
                 && contextValue.fetchForAddress !== accounts.current.address
                 && isCurrentAuthenticated()) {
             const currentAccount = accounts!.current!.address;
-            const refreshRequests = (clear: boolean) => refreshRequestsFunction(clear, currentAccount, dispatch, axiosFactory);
-            const axios = axiosFactory(currentAccount);
-            const updateSettingCallback = async (id: string, value: string) => {
-                await updateSetting(axios, id, value);
+            const currentLegalOfficer = client.allLegalOfficers.find(legalOfficer => legalOfficer.address === currentAccount);
```

Comment:
Any reason for not using
```suggestion
            const currentLegalOfficer = getOfficer(currentAccount);
```
?

Reply:
None indeed, applied!